File: Python3/Medium/WrongAnswer/string-compression_6.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        
        i, j = 0, 0
        while i < len(chars):
            c = chars[i]
            while j < len(chars) and chars[j] == c:
                j += 1
            if j - i > 1:
                for k in str(j - i):
                    chars[i + 1] = k
                    i += 1
            i += 1
        return i

File: Java/Medium/Accepted/string-compression_6.java
class Solution {
    public int compress(char[] chars) {
        
        int n = chars.length;
        int count = 1;
        int idx = 0;
        for (int i = 0; i < n; i++) {
            if (i < n - 1 && chars[i] == chars[i + 1]) {
                count++;
            } else {
                chars[idx++] = chars[i];
                if (count > 1) {
                    String cnt = String.valueOf(count);
                    for (char c : cnt.toCharArray()) {
                        chars[idx++] = c;
                    }
                }
                count = 1;
            }
        }
        return idx;
    }
}